pallets/confidential/src/lib.rs

```diff
@@ -79,6 +88,36 @@ decl_module! {
 
         fn deposit_event() = default;
 
+        #[weight = 400_000]
+        pub fn store_valid_asset_ids(origin, // TODO: how to ensure only privileged users can call this?
+            valid_asset_ids: Vec<AssetId>,
+        ) -> DispatchResult
+        {
+            ensure_signed(origin)?;
+
+            <ValidAssetIds>::put(valid_asset_ids);
+            Ok(())
+        }
+
+        #[weight = 400_000]
+        pub fn store_account_tx(origin,
+            tx: PubAccountTx,
+        ) -> DispatchResult
+        {
+            let owner_acc = ensure_signed(origin)?;
+            let owner_id = Context::current_identity_or::<Identity<T>>(&owner_acc)?;
+
+            let account_vldtr = AccountValidator {};
+            let valid_asset_ids = Self::valid_asset_ids();
+            let valid_asset_ids = convert_asset_ids(valid_asset_ids);
```

Comment:
Right now I am calling `convert_asset_ids` to convert them from `AssetId` type to `Scalar` type.
- One reason was that storing `Scalar` on the chain would not work since `Scalar` does not implement Encode/Decode.
- The other reason is that I am not yet sure about how we will end up converting plain ticker names to asset ids.

Reply:
I am assuming the asset_ids are unique and encrypted in some sense so no one can know which asset ticker has which asset_id.

FYI - We can use the asset Identity as well that is generated on the basis of the hash(SecurityToken + Ticker) if assetId is not encrypted.

Reply:
Yes, the asset_ids of MERACT lib are encrypted and unique. Great, asset identity works well.